### PR TITLE
Added Reducers and Actions and Modified Routes

### PR DESCRIPTION
**Pull request by becauseIknowNothing**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
 import { Switch, BrowserRouter, Route, Link } from 'react-router-dom';
-
 import HomePage from './components/HomePage/HomePage';
 import AboutUs from './components/AboutUs/AboutUs';
 import Guests from './components/Guests/Guests';
 import Contact from './components/Contact/Contact';
 import Sponsors from './components/Sponsors/Sponsors';
 import User from './components/User/User';
 import LoggedIn from './LoggedIn';		
+import NotFound from './components/NotFound/NotFound'
```
Comment (ciphereck) [inline src/App.js]: change name for NotFound. Not very intutive. You can use 404NotFound

File: src/reducers/gallery.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            return action.payload.data.images
+        case 'FETCH_GALLERY_FAILURE':
+            //ReactModal to be used as popup
+            return state;
+        default:
+            return state;
+    }
+}
+
+export const IsFetchingGallery = (state = initialIsFetchingGallery, action) => {
```
Comment (ciphereck) [inline src/reducers/gallery.js]: function name should be camel Case.
isFetchingGallery

File: src/store/store.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import thunk from 'redux-thunk';
+import { createStore, applyMiddleware, compose} from 'redux';
+import rootReducer from '../reducers';
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+export default (preloadedState) => {
+    const store = createStore(
+        rootReducer,
+        preloadedState,
+        composeEnhancers(applyMiddleware(thunk))
```
Comment (ciphereck) [inline src/store/store.js]: also add loggerMiddleware for now

File: src/reducers/gallery.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
+const initialGalleryState = [];
+const initialIsFetchingGallery = false
+export const gallery = (state = initialGalleryState, action) => {
+    switch (action.type) {
+        case 'FETCH_GALLERY_SUCCESS':
```
Comment (ciphereck) [inline src/reducers/gallery.js]: create another file actionTypes.js
declare all string constants in that file
all other files should refer to those strings for action types

File: src/actions/gallery.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
+export const successGallery = (payload) => ({
```
Comment (ciphereck) [inline src/actions/gallery.js]: how exactly are you using these actions
you need to create services for api calls and then use actions in components to make api calls.
What is the purpose of this function ?

File: src/App.js (hunk 1 above)
Comment (becauseIknowNothing) [reply src/App.js]: `An identifier or keyword cannot immediately follow a numeric literal.`
Can't use 404NotFound, how about Error404NotFound?

Comment (ankurcharan) [reply src/App.js]: fine !

**Pull request merged**